pysph/sph/wc/linalg.py

```diff
@@ -21,6 +21,56 @@ def dot(a=[0.0, 0.0], b=[0.0, 0.0], n=3):
     return result
 
 
+def deteminant_2d(A=[0.0, 0.0]):
+    return A[0]*A[3] - A[1]*A[2]
+
+
+def deteminant_3d(A=[0.0, 0.0]):
+    c1 = deteminant_2d([A[4], A[5], A[7], A[8]])
+    c2 = deteminant_2d([A[3], A[5], A[6], A[8]])
+    c3 = deteminant_2d([A[3], A[4], A[6], A[7]])
+    result = A[0] * c1 -  A[1] * c2 + A[2] * c3
+    return result
+
+
+def deteminant_4d(A=[0.0, 0.0]):
+    c1 = deteminant_3d([A[5], A[6], A[7],
+                   A[9], A[10], A[11],
+                   A[13], A[14], A[15]])
+    c2 = deteminant_3d([A[4], A[6], A[7],
+                   A[8], A[10], A[11],
+                   A[12], A[14], A[15]])
+    c3 = deteminant_3d([A[4], A[5], A[7],
+                   A[8], A[9], A[11],
+                   A[12], A[13], A[15]])
+    c4 = deteminant_3d([A[4], A[5], A[6],
+                   A[8], A[9], A[10],
+                   A[12], A[13], A[14]])
+    result = A[0] * c1 - A[1] * c2 + A[2] * c3 - A[3] * c4
+    return result
```

Comment:
Big trouble here! If I pass `A` which is defined as `A = matrix('9')` and then intialized as did in the `kernel_corrections.py` file, with `dim = n = 2` and if I call `deteminant_2d(A, b, res)` where `A` is a 9 element 1-D array with only `1, 2, 4, 5` values of the array are initialized to the correct values and the remaining `3, 6, 7, 8, 9` values possibly undefined. `deteminant_2d` will give me the wrong value!

Reply:
Added a check for diagonal elements, if it is zero than the solver return 0.0 instead 1.0 with the result.